pydicer/preprocess/data.py

```diff
@@ -1,4 +1,73 @@
+from pathlib import Path
+import pydicom
+import numpy as np
+
+
 class PreprocessData:
-    def preprocess():
+    """
+    Class for preprocessing the data information into a dicionary that holds the data in a
+    structured hierarchy
+
+    Args:
+        working_directory (Path): The working directory in which the data is stored (Output of the
+        Input module)
+    """
+
+    def __init__(self, working_directory):
+        self.working_directory = working_directory
+
+    # TODO: need to find the linked series UID
+    def preprocess(self):
+        """
+        Function to preprocess information regarding the data located in an Input working directory
+
+        Returns: res_dict (Dict): keys are series UIDs. For each series; we have 5 lower keys:
+            - the hashed patient ID
+            - the hashed study UID
+            - a list of dicts that has 2 keys (path to file, slice location)
+            - the series modality
+            - the series UID that it is linked to
+
+        Ex:
+            {
+                "series_uid": {
+                    "patient_id": "",
+                    "study_id": "",
+                    "files": [],
+                    "modality": "",
+                    "linked_series_uid": ""
+                }
+            }
+        """
```

Comment:
Might just have to check how the formatting of this docstring ends up in the sphinx docs... We can worry about that later though.